lib/ddq-backend-mysql.js

```diff
@@ -331,8 +331,10 @@ module.exports = (configValidation, crypto, EventEmitter, mysql, timers) => {
         /**
          * Resets the relevant flags and clears the timers, which stops both
          * polling and restoring.
+         *
+         * @param {Function} [callback]
          */
-        pausePolling() {
+        stopListening(callback) {
```

Comment:
This function, due to a race condition, can complete and call the callback.  After successfully `stopListening` the ddq core could get an event with a wrapped message.

To close this hole, you'll have to wrap the emit in a condition checking if you are still polling.  But I don't like that solution.  I would rather have stopListening stall until we know it is certainly stopped, especially if we intend to close the connection as well.  ... Thinking.